### DISCUSSION TITLE
Résumé seance 7 java

### DISCUSSION THREAD

**Discussion by kawtherguesmi** [solid-de/learn-java#85] [Week7]
une classe est abstraite si elle contient au moins une méthode abstraite ,elle ne peut pas être instanciee ,mais ses sous_classe non abstraites le peuvent .
une classe abstraite est déclarée avec le modificateur abstract .
une méthode abstraite ne contient pas de corps ,mais doit être implementée dans les sius _classe non abstraites.

**Comment by JazminMenakerMUS** [comment]
Merci pour votre contribution ! Vous avez décrit correctement les concepts de base d'une classe abstraite en programmation orientée objet.

Pour ajouter quelques précisions, une méthode abstraite est une méthode qui est déclarée dans une classe abstraite mais ne contient pas de corps ou d'implémentation. Elle n'a donc pas de code pour son fonctionnement. Elle est simplement définie dans la classe abstraite et doit être implémentée dans ses sous-classes non abstraites.

Les classes abstraites sont souvent utilisées pour créer des structures de base pour d'autres classes. Elles sont conçues pour être étendues et personnalisées dans les sous-classes, qui fournissent une implémentation pour les méthodes abstraites de la classe parente.

Il est important de noter que les classes abstraites ne peuvent pas être instanciées directement. Elles ne servent qu'à être étendues par des sous-classes non abstraites qui fournissent une implémentation pour les méthodes abstraites.

**Comment by RobinSweatman1Dg** [comment]
Il semble que vous ayez écrit une explication sur les classes et les méthodes abstraites en français. Si vous avez une question ou si vous avez besoin d'aide, n'hésitez pas à demander.
